enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/lock/forseti/SharedLock.java

```diff
@@ -214,6 +215,26 @@ public String describeWaitList()
     }
 
     @Override
+    public void collectOwners( Set<ForsetiClient> owners )
+    {
+        for ( AtomicReferenceArray<ForsetiClient> ownerArray : clientsHoldingThisLock )
```

Comment:
Yes, and I think this is the reason that, as far as I have observed, deadlocks that involve shared locks are still the still the most probable false positive to see.